### PR TITLE
Adding NemoClient for retrieving data from NeMO Identifier API (SCP-5226)

### PR DESCRIPTION
**Pull request by bistline**
#### BACKGROUND & CHANGES
This update adds the `NemoClient` for retrieving data from the forthcoming NeMO Identifier API.  This API is still in a beta release state, and is behind basic auth until it goes public later this year.  This client is not integrated anywhere yet, but has bindings for all the major entity types: files, projects, publications, etc.

NOTE:  This PR does not validate that we can retrieve all required information from the NeMO Identifier API.  Almost assuredly, there will be information missing that SCP requires, and this will need to be handled by the yet-to-be-implemented `ImportService`.  This update only deals with API connectivity.  But we will be able to retrieve file names/types/access urls, and basic project attributes like names, associated publications, etc.  Many file-level attributes will have to be set from default configurations as they may not be available (e.g. `library_preparation_protocol` for expression data or `obsm_key_name` for AnnData clustering embeds).

You can also interact with the [beta API](https://beta-assets.nemoarchive.org) in a browser, using the credentials from vault (look for `NEMO_API_USERNAME` and `NEMO_API_PASSWORD`).

#### MANUAL TESTING
1. Load a fresh copy of environment secrets and enter a Rails console session
2. Instantiate the client and validate it loads credentials and can see the API
```
client = NemoClient.new
=> 
#<NemoClient:0x0000000106e6e018

client.username
=> "single-cell-portal"

client.api_available?
=> true
```
3. Load an entity, and use the association links provided to traverse the data structures:
```
file_id = 'nemo:der-ah1o5qb'
=> "nemo:der-ah1o5qb"

file = client.file(file_id)
file
=> 
{"id"=>"nemo:der-ah1o5qb",                              
 "file_name"=>"human_var_scVI_VLMC.h5ad.tar",           
 "md5"=>"451a28b7f0f809335ba9d78a720ebc9b",             
 "sha256"=>nil,                                                                     
 ...

collection_id = file['collections'].first.split('/').last
=> "nemo:col-hwmwd2x"

collection = client.collection(collection_id)
collection
=> 
{"id"=>"nemo:col-hwmwd2x",           
 "col_type"=>"files",                
 "is_static"=>0,                     
 "short_name"=>"human_variation_10X",
 "name"=>"\"Human variation study (10x), GRU\"",
 ...
 
project_id = collection['projects'].first.split('/').last
=> "nemo:grn-gyy3k8j"

project = client.project(project_id)
project
=> 
{"id"=>"nemo:grn-gyy3k8j", 
 "project_type"=>"grant",
 "program"=>"other",
 "short_name"=>"AIBS Internal",
 "title"=>"Projects funded by internal Allen Institute for Brain Science funding",
 ...
```

### CODE REVIEW

File: app/models/nemo_client.rb [hunk 1]
```diff
@@ -0,0 +1,221 @@
+# API client for interacting with Neurpscience Multi-Omic Data Portal (NeMO)
```
Comment (eweitz) [inline app/models/nemo_client.rb:1]:
```suggestion
# API client for interacting with Neuroscience Multi-Omic Data Portal (NeMO)
```

File: app/models/nemo_client.rb [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+      status = execute_http_request(:get, path)&.with_indifferent_access
+      status && status[:status] == 'OK'
+    rescue RestClient::ExceptionWithResponse => e
+      Rails.logger.error "NeMO service unavailable: #{e.message}"
+      ErrorTracker.report_exception(e, nil, { method: :get, url: path, code: e.http_code })
+      false
+    end
+  end
+
+  # generic handler to get an entity from the NeMO identifier API
```
Comment (eweitz) [inline app/models/nemo_client.rb:116]:
```suggestion
  # generic handler to get an entity from the NeMO Identifier API
```

**Review by eweitz — APPROVED** [review]
It's good to see this starting up!  Code looks good, and manual test works as expected.

**Review by jlchang — APPROVED** [review]
Functional review: works as described.

Glad you were finally unblocked on this!

**Pull request merged**